cmd/minikube/cmd/start_flags.go

```diff
@@ -571,6 +571,10 @@ func updateExistingConfigFromFlags(cmd *cobra.Command, existing *config.ClusterC
 	updateBoolFromFlag(cmd, &cc.KubernetesConfig.ShouldLoadCachedImages, cacheImages)
 	updateIntFromFlag(cmd, &cc.KubernetesConfig.NodePort, apiServerPort)
 
+	if cmd.Flags().Changed(kubernetesVersion) {
```

Comment:
Nice catch ! Is there a chance that we missed more changed flags ?

Reply:
Thanks! There could be. Let me do a thorough check.

Reply:
I did find `apiserver-names` being removed as well: https://github.com/kubernetes/minikube/commit/50637dd1498504c1f15aa911319cd148c963633e#diff-0e864ab4025634664724909a47c34fbcae246ad52307eaaaa58153f0b256a8b4L644-L646
So I added it in 7b98243ea31c214de4f7639249b7d65991aaf7f3.
Ideally I think we should cover this piece of logic in the unit test. Might need some refactoring. Will look into it and probably open a follow-up PR.

Thanks!

Reply:
@dinever yes  please a refactor to make this simplified is much needed !

I really like your contributions ! would you be interested to join our office hours sometimes ? if it works with your timezone?
https://minikube.sigs.k8s.io/community/

Reply:
Thanks @medyagh! Yep I'd love to join the office hours. I've added it to my calendar. Thanks!